pkg/workload/tpcc/timeutil.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tpcc
+
+import (
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
+	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
+	"golang.org/x/exp/rand"
+)
+
+type tpccTime struct {
+	syncutil.Mutex
+	fakeTime time.Time
+	stepMax  time.Duration
+	rng      *rand.Rand
+}
+
+func (t *tpccTime) Now() time.Time {
+	if t.fakeTime.IsZero() {
+		return timeutil.Now()
+	}
+
+	t.Lock()
+	defer t.Unlock()
+	t.fakeTime = t.fakeTime.Add(
+		time.Duration(max(t.rng.Float64()*t.stepMax.Seconds(), 1)) * time.Second,
+	)
+	return t.fakeTime
+}
+
+func newTpccTime(fakeStartTime time.Time, stepMax time.Duration, seed uint64) *tpccTime {
+	if fakeStartTime.IsZero() {
```

Comment:
nit: i think this `if fakeStartTime.IsZero() {`  branch should gate all of the logic at the beginning of `worker.run()`, as  it would make it clear to the reader that using this fake time is non default. In other words, `newTpccTime()` always creates a legit fake clock.

Reply:
Oh that's a good idea, I'll update to that